python/taichi/lang/meta.py

```diff
@@ -16,13 +16,24 @@ def tensor_to_ext_arr(tensor: ti.template(), arr: ti.ext_arr()):
 
 
 @ti.kernel
-def vector_to_fast_image(img: ti.template(), out: ti.ext_arr()):
+def vector_to_fast_image(img: ti.template(), small_endian: ti.template(),
+                         out: ti.ext_arr()):
     # FIXME: Why is ``for i, j in img:`` slower than:
     for i, j in ti.ndrange(*img.shape):
         u, v, w = min(255, max(0, int(img[i, img.shape[1] - 1 - j] * 255)))
+        idx = j * img.shape[0] + i
         # We use i32 for |out| since OpenGL and Metal doesn't support u8 types
-        # TODO: treat Cocoa and Big-endian machines, with XOR logic
-        out[j * img.shape[0] + i] = w + (v << 8) + (u << 16)
+        if ti.static(small_endian):
```

Comment:
```suggestion
        if ti.static(ti.get_os_name() != 'osx'):
```
instead of passing a template argument?
Btw, can you confirm if all OS X machines are big-endian (and also all Win and Linux X11 are little-endian)?

Reply:
Fixed.

>  can you confirm if all OS X machines are big-endian

Given that the non-fast way is using big-endian and we don't hear users complain, I think this is safe. Let's just roll it out and see what happens, shall we?